client/src/containers/widget/horizontal-bar-chart/sandbox/index.tsx

```diff
@@ -0,0 +1,98 @@
+"use client";
+import React, { FC, useRef, useEffect, useState, useCallback } from "react";
+
+import { WidgetChartData } from "@shared/dto/widgets/base-widget-data.interface";
+import { useDebouncedCallback } from "use-debounce";
+
+import BaseHorizontalBarChart from "@/containers/widget/horizontal-bar-chart";
+
+interface LargeHorizontalBarChartProps {
+  data?: WidgetChartData;
+}
+
+const getTextWidth = (text: string) => {
+  const canvas = document.createElement("canvas");
+  const context = canvas.getContext("2d");
+  return context ? context.measureText(text).width : 0;
+};
+
+const LEFT_PADDING = 24;
+
+const HorizontalBarChart: FC<LargeHorizontalBarChartProps> = ({ data }) => {
+  const height = (data && data.length * 50) || 0;
+  const chartRef = useRef<HTMLDivElement>(null);
+  const [chartWidth, setChartWidth] = useState(0);
+  const updateChartWidth = useCallback(() => {
+    if (chartRef.current) {
+      setChartWidth(chartRef.current.offsetWidth);
+    }
+  }, []);
+
+  const debouncedUpdateChartWidth = useDebouncedCallback(updateChartWidth, 250);
+
+  useEffect(() => {
+    updateChartWidth();
+    window.addEventListener("resize", debouncedUpdateChartWidth);
+    return () =>
+      window.removeEventListener("resize", debouncedUpdateChartWidth);
+  }, [debouncedUpdateChartWidth]);
```

Comment:
Have you tried wrapping the chart around `ResponsiveContainer`? https://recharts.org/en-US/api/ResponsiveContainer